### DISCUSSION TITLE
Improve search with tagging/categorization

### DISCUSSION THREAD

**Discussion by toobulkeh** [rubygems/rubygems.org#3353] [Ideas]
Based on @dwradcliffe's comment on [this Ruby Toolbox](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-309128895) discussion, I'm interested in starting a discussion on categorization / tagging with RubyGems itself.

I know GitHub has [recently started supporting "topics"](https://github.com/blog/2309-introducing-topics), which may be a natural place to pull / coordinate from, though it's [only available over the API in preview form](https://developer.github.com/v3/repos/).

**Comment by ghost** [comment]
I am all for it (provided that people don't do stupid things with tags).

I assume that with the new search filter/options, the projects with more downloads may appear first, so perhaps my worry about inappriopriate tags is wrongful.

And don't get me wrong - I am in support of this; in general, tagging helps a LOT in so many ways (if the tags make sense).

In one of my projects where I compile stuff from source, I also added a yaml file to group different programs into different categories. This is not always easy to do, and neither is picking the right tags... but for example, a simple one is "programming_language" tag for stuff like ruby, python etc...

It gets a bit harder when I use group tags such as multimedia, video, audio. Bit difficult to say where things are distinguished.

Perhaps if there is some limiter... so that people don't go overboard with like using 100 random tags. Pick something like... I dunno... 0-1, 0-4, or perhaps up to 10 or so - should cover most; and if someone really needs more than 10 well they could always file an issue request. :)

I liked the toolbox but I understand that it would be too much work for him.

I think if rubygems could integrate a "lightweight" variant then this would be neat. Like, keep the tags optional, so search/filter may still work anyway even without tags. But tags may help when filtering even more.

Anyway, sorry for the length - +1 from me here.

**Comment by simi** [comment]
There is no way to get gem tags/category currently. In theory that could be added to specification and pulled on gem push. But that would need a lot of work on both sides (RubyGems itself and RubyGems.org UI/API). If anyone is interested, feel free to open RFC or initial PR to RubyGems to incorporate optional specification fields (anyway I can't guarantee that would be welcomed in specification, that's just an initial idea). Moving to discussions for now.
